frame/support/src/traits.rs

```diff
@@ -1185,6 +1185,79 @@ pub trait VestingSchedule<AccountId> {
 	fn remove_vesting_schedule(who: &AccountId);
 }
 
+/// Abstraction over a fungible asset system.
+pub trait FungibleAsset<AccountId> {
+	/// The balance of an account.
+	type Balance: AtLeast32BitUnsigned + FullCodec + Copy + MaybeSerializeDeserialize + Debug +
+		Default + HasCompact;
+
+	type AssetId: Member + Parameter + Default + Copy + HasCompact;
+
+	// PUBLIC IMMUTABLES
+	/// The total amount of issuance in the system for a specific asset.
+	fn total_issuance(id: Self::AssetId) -> Self::Balance;
+
+	/// The balance of a given account.
+	///
+	/// This is the only balance that matters in terms of most operations on tokens. It alone
+	/// is used to determine the balance when in the contract execution environment. When this
+	/// balance falls below the value of `ExistentialDeposit`, then the 'current account' is
+	/// deleted: specifically `FreeBalance`.
+	///
+	/// `system::AccountNonce` is also deleted if `ReservedBalance` is also zero (it also gets
+	/// collapsed to zero if it ever becomes less than `ExistentialDeposit`.
+	fn balance_of(id: Self::AssetId, who: AccountId) -> Self::Balance;
+
+	// PUBLIC MUTABLES (DANGEROUS)
+
+	/// Transfer some liquid free balance of an asset to another account.
+	fn transfer(id: Self::AssetId, from: AccountId, to: AccountId, amount: Self::Balance) -> DispatchResult;
+}
+
+pub trait MintableAsset<AccountId>: FungibleAsset<AccountId> {
+	/// Increase the total issuance of of a specific asset by `amount` for a specific account.
+	///
+	/// Returns `Ok` iff the mint was successful.
```

Comment:
this feels redundant

Reply:
I think it’s useful for readers to have docs of return values.